### DISCUSSION TITLE
Unable to compress b64 images

### DISCUSSION THREAD

**Discussion by ArunNetstager** [spatie/image-optimizer#169] [Q&A]
I have used a Jquery Image cropper in the frontend. So I get b64 code in my laravel controller and file saved to my storage location.

Then I'm trying to reduce my file size with my file's storage path.

Unfortunately, It's not working...!

Here is my codes in controller

               

`
               $data = base64_decode($imageArray2[1]);

$imageName = time().'_'.Str::random(40).'.png';
Storage::disk('public')->put('admin/article/image/'.$imageName, $data);
app(OptimizerChain::class)->optimize(storage_path('app/public/admin/article/image/'.$imageName));`

**Comment by Tyjow** [comment]
Hi guys,

Your path can work but when you check it with `is_file()` PHP function it return **false**
It must be return **true** to be a "valid" path
To get a valid path do :

```
$data = base64_decode($imageArray2[1]);
$imageName = time().'_'.Str::random(40).'.png';

Storage::disk('public')->put('/admin/article/image/' . $imageName, $data);

$pathToImage = Storage::disk('public')->getAdapter()->applyPathPrefix('/admin/article/image/' . $imageName);
app(OptimizerChain::class)->optimize($pathToImage );
```

if your image exist it should work !
